mobile-app/app/components/DomainSwitch.tsx

```diff
@@ -0,0 +1,112 @@
+import { View, Text } from "react-native";
+import { ThemedTouchableOpacityV2 } from "@components/themed";
+import { getNativeIcon } from "@components/icons/assets";
+import { useThemeContext } from "@waveshq/walletkit-ui";
+import React from "react";
+import { useDomainContext, DomainType } from "@shared-contexts/DomainProvider";
+import { DomainPersistence } from "@api";
+import { tailwind } from "@tailwind";
+import { LinearGradient } from "expo-linear-gradient";
+
+export function DomainSwitch({ testID }: { testID: string }): JSX.Element {
+  const { isLight } = useThemeContext();
+  const { domain, setDomain } = useDomainContext();
+  const isEvmDomain = domain !== DomainType.DFI;
+  const DFIIcon = getNativeIcon("DFIlogo");
+  const EvmDFIIcon = getNativeIcon("EvmDFI");
+
+  return (
+    <View
+      style={tailwind("h-7 w-18 rounded-full flex-row justify-center mr-4", {
+        "bg-mono-light-v2-300": isLight,
+        "bg-mono-dark-v2-300": !isLight,
+      })}
+      testID={testID}
+    >
+      <ThemedTouchableOpacityV2
```

Comment:
hmm thinking if we should just do something like this to reduce `isEvmDomain` conditions and css show/hide
```
{domain === DomainType.DFI && (<ThemedTouchableOpacityV2 ....>....</ThemedTouchableOpacityV2>)
{domain === DomainType.EVM && (<ThemedTouchableOpacityV2 ....>....</ThemedTouchableOpacityV2>)
```